Tools/TestWebKitAPI/Tests/WebKitCocoa/ResourceLoadStatistics.mm

```diff
@@ -2201,3 +2202,39 @@ HTTPServer httpServer({
     EXPECT_FALSE(didSendSite2CookieHeader);
     didSendSite2CookieHeader = false;
 }
+
+TEST(ResourceLoadStatistics, LoginStatusRequiresPasswordAutoFill)
+{
+    TestWebKitAPI::HTTPServer server({
+        { "/index"_s, { @"" } }
+    });
+    RetainPtr configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+
+    for (_WKFeature *feature in WKPreferences._features) {
+        if ([feature.key isEqualToString:@"LoginStatusAPIEnabled"] || [feature.key isEqualToString:@"LoginStatusAPIRequiresWebAuthnEnabled"])
+            [[configuration preferences] _setEnabled:YES forFeature:feature];
+    }
+    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 100, 100) configuration:configuration.get()]);
+
+    [webView loadRequest:server.request("/index"_s)];
+    [webView _test_waitForDidFinishNavigation];
+    [webView _didFillPasswordForUsername:@"example"];
+
+    __block bool done { false };
+    [webView evaluateJavaScript:@"navigator.setStatus('logged-in');" completionHandler:^(id, NSError *) {
```

Comment:
Any chance you can move these under the navigator.login type?

```suggestion
    [webView evaluateJavaScript:@"navigator.login.setStatus('logged-in');" completionHandler:^(id, NSError *) {
```

For reference, here is what we have implemented in Chrome:

https://w3c-fedid.github.io/login-status/#dom-navigatorlogin-setstatus

Reply:
@johnwilander FYI